### PR TITLE
Updating GC with API pagination and revision tags.

### PR DESCRIPTION
**Pull request by mohammedzee1000**
https://github.com/CentOS/container-pipeline-service/issues/515

- [x] Adding pagination for querying registry API for catalog.
- [x] Adding ability to clean up revision tags.

### CODE REVIEW

File: container_pipeline/lib/shell/registry_delete_revision_tags.sh [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+fi
+
+REG_PATH="/var/lib/registry/docker/registry/v2/repositories/";
+MANIFEST_LIST="/tmp/registry_manifests.txt"
+
+pushd ${REG_PATH}
+
+echo "Searching for revision tags to remove ... ";
+
+MANIFESTS_WITHOUT_TAGS=$(comm -23 <(find . -type f -name "link" | grep "_manifests/revisions/sha256" | grep -v "\/signatures\/sha256\/" | awk -F/ '{print $(NF-1)}' | sort) <(for f in $(find . -type f -name "link" | grep "_manifests/tags/.*/current/link"); do cat ${f} | sed 's/^sha256://g'; echo; done | sort))
```
Comment (navidshaikh) [inline container_pipeline/lib/shell/registry_delete_revision_tags.sh]: should you simplify this operation? or decorate a complex operation with comments about whats being done here?

File: container_pipeline/utils.py [hunk 2]
```diff
@@ -1,12 +1,12 @@
...
 import json
-import yaml
 import logging
 import os
-import urllib2
 import subprocess
+import urllib2
 from shutil import rmtree
 
+import yaml
```
Comment (navidshaikh) [inline container_pipeline/utils.py]: you can group direct `import` and `from ` imports

File: container_pipeline/utils.py [hunk 3]
```diff
@@ -33,27 +33,45 @@ def load_yaml(yaml_file):
     return data
 
 
-def run_cmd(cmd, check_call=True, no_shell=False):
+def run_cmd(cmd, check_call=True, no_shell=False, use_shell=False,
```
Comment (navidshaikh) [inline container_pipeline/utils.py]: why repeat the utilities code? check `container-pipeline-service/container_pipeline/lib/command.py` add necessary options in that utility function with proper defaults (to keep up with other code already using it)

Comment (mohammedzee1000) [reply container_pipeline/utils.py]: That is something we need to do, consolidate into one lib. We have various run_cmd and other such funcs scattered accross the board, which i too dont like. Maybe a seperate PR is needed.

**Pull request merged**